### PR TITLE
feat(provider): add smsmode provider

### PR DESCRIPTION
**Pull request by nohira**
### What change does this PR introduce?

Add [smsmode](https://www.smsmode.com/) provider implementation

### Why was this change needed?

This provider is used in my product

### CODE REVIEW

**Comment by nohira** [comment]
> @nohira Could you look into merge conflicts please

I've rebase onto next and resolved conflict due to a new sms provider previously merged

**Comment by nohira** [comment]
I've updated the project after brevo-sms PR merge

File: providers/smsmode/src/lib/smsmode.provider.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+} from '@novu/stateless';
+import { ProxyAgent } from 'proxy-agent';
+import 'cross-fetch';
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  interface RequestInit {
+    agent: ProxyAgent;
+  }
+}
```
Comment (jainpawan21) [inline providers/smsmode/src/lib/smsmode.provider.ts:15]: @scopsy shall we allow `proxy-agent` and `cross-fetch` in place of `axios`?

Comment (nohira) [reply providers/smsmode/src/lib/smsmode.provider.ts:15]: I used this two for avoiding 502 proxy issue that axios have. You can see my other PR about proxy on sendinblue. Also, I've used it in brevo sms provider.

**Pull request closed without merging**